src/com/owncloud/android/utils/ErrorMessageAdapter.java

```diff
@@ -92,6 +93,9 @@ public static String getErrorCauseMessage(RemoteOperationResult result,
                 } else if (result.getCode() == ResultCode.INVALID_CHARACTER_DETECT_IN_SERVER) {
                     message = res.getString(R.string.filename_forbidden_charaters_from_server);
 
+                } else if (result.getCode() == ResultCode.MAINTENANCE_MODE) {
```

Comment:
Instead of adding the case of ```ResultCode.MAINTENANCE_MODE``` to every operation block, I would just add it to ```getErrorMessage()```  and ```isNetworkError()```. Maybe the methods of these two methods could be changed to ```getCommonErrorMessage()```  and ```isCommonError()``` to show better their intention.